activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -112,20 +112,27 @@ def table(table, stream)
           tbl = StringIO.new
 
           # first dump primary key column
-          pk = @connection.primary_key(table)
+          pk = @connection.primary_keys(table) rescue nil
```

Comment:
`primary_keys` is new internal API for correctly dump/load composite primary key.
If another adapter (e.g. [oracle-enhanced](https://github.com/rsim/oracle-enhanced), [activerecord-sqlserver-adapter](https://github.com/rails-sqlserver/activerecord-sqlserver-adapter)) is not implemented this new API, raise a `NotImplementedError`.